### PR TITLE
added DTO, Mapstruct, mappers, BookController, EntityNotFoundExceptio…

### PR DESCRIPTION
**Pull request by AgroFix**
…n class, added new methods (getAll(), getBookById, CreateBookRequest), remove CommandLineRunner bean

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java [hunk 1]
```diff
@@ -42,6 +43,16 @@ public Book save(Book book) {
         return book;
     }
 
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.get(Book.class, id);
+            return Optional.ofNullable(book);
```
Comment (nshtykh) [inline src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java]: You don't need to create an extra variable - return Optional.ofNullable(session.get(Book.class, id));

**Review by nshtykh — APPROVED** [review]
Great job!

File: src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/mate/academy/bookstore/repository/BookRepositoryImpl.java]: Makes sense.

**Review by vovasalyha — APPROVED** [review]
LGTM

**Pull request merged**